x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusReindexedIntoStoredSourceChallengeRestIT.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.logsdb.qa;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * This test compares behavior of a standard mdoe data stream and a logsdb data stream using stored source.
+ * There should be no differences between such two data streams.
+ */
+public class LogsDbVersusReindexedIntoStoredSourceChallengeRestIT extends ReindexChallengeRestIT {
+    public String getBaselineDataStreamName() {
+        return "logs-apache-baseline";
+    }
+
+    public String getContenderDataStreamName() {
+        return "logs-apache-reindexed";
+    }
+
+    @Override
+    public void baselineSettings(Settings.Builder builder) {
+        dataGenerationHelper.logsDbSettings(builder);
+    }
+
+    @Override
+    public void contenderSettings(Settings.Builder builder) {
```

Comment:
I am wondering if we should just create base classes like

```
SyntheticSourceAbstractChallengeRestTest
DisableSourceAbstractChallengeRestTest
StoredSourceAbstractChallengeRestTest
```

where the setting is applied and we can just inherit from them.

Reply:
The thing is that sometimes baseline has stored source and sometimes the contender.